jax/experimental/array_serialization/serialization.py

```diff
@@ -185,6 +187,20 @@ async def release_bytes(self, requested_bytes):
       self._cv.notify_all()
 
 
+async def transfer_shard_to_host(shard: array.Shard) -> np.ndarray:
+  data = shard.data
+  has_pinned_host = any(
+      m.kind == "pinned_host" for m in shard.device.addressable_memories())
+  if config.enable_memories.value and has_pinned_host:
+    # If available, transfer to pinned host memory
+    sharding = jax.sharding.SingleDeviceSharding(shard.device,
+        memory_kind="pinned_host")
+    data = jax.jit(lambda x: x, out_shardings=sharding)(data)
+    # Allow other transfers to be scheduled simultaneously
+    await asyncio.sleep(0)
```

Comment:
Why do we need this sleep? `await` should schedule it concurrently anyways right?

Reply:
(Deferred to a separate commit)